bpf/lib/common.h

```diff
@@ -520,6 +520,48 @@ enum {
 	CT_RELATED,
 };
 
+/*
+ * SVC_FLAG_*_: Service flags (lb{4,6}_service->flags)
+ *
+ * We use SVC_CHECK_* for checks so that they can be optimized out if a flag is
+ * not enabled (see ENABLE_* defines).
+ */
+enum {
+	SVC_FLAG_EXTERNAL_IP_  = (1 << 0),  /* K8s External IPs */
+	#if defined(ENABLE_EXTERNAL_IP)
+	#define SVC_CHECK_EXTERNAL_IP SVC_FLAG_EXTERNAL_IP_
+	#else
+	#define SVC_CHECK_EXTERNAL_IP 0
+	#endif
+
+	SVC_FLAG_NODEPORT_     = (1 << 1),  /* K8s NodePort service */
+	#if defined(ENABLE_NODEPORT)
+	#define SVC_CHECK_NODEPORT SVC_FLAG_NODEPORT_
+	#else
+	#define SVC_CHECK_NODEPORT 0
+	#endif
+
+	SVC_FLAG_LOCAL_SCOPE_  = (1 << 2),  /* K8s externalTrafficPolicy=Local */
+	#define SVC_CHECK_LOCAL_SCOPE SVC_FLAG_LOCAL_SCOPE_
+
+	SVC_FLAG_HOSTPORT_     = (1 << 3),  /* K8s hostPort forwarding */
+	#if defined(ENABLE_HOSTPORT)
+	#define SVC_CHECK_HOSTPORT SVC_FLAG_HOSTPORT_
+	#else
+	#define SVC_CHECK_HOSTPORT 0
+	#endif
+
+	SVC_FLAG_AFFINITY_     = (1 << 4),  /* K8s sessionAffinity=clientIP */
+	#define SVC_CHECK_AFFINITY SVC_FLAG_AFFINITY_
+
+	SVC_FLAG_LOADBALANCER_ = (1 << 5),  /* K8s LoadBalancer service */
+	#if defined(ENABLE_LOADBALANCER)
+	#define SVC_CHECK_LOADBALANCER SVC_FLAG_LOADBALANCER_
+	#else
+	#define SVC_CHECK_LOADBALANCER 0
+	#endif
```

Comment:
I would rather prefer to stick to kernel style, meaning always have the constants defined to their expected value, and then for e.g. `lb4_svc_is_loadbalancer()` do something like: `return is_defined(ENABLE_LOADBALANCER) (svc->flags & SVC_CHECK_LOADBALANCER);`

The `is_defined()` will either resolve to 0 or 1 at compile time, so it will have the same effect, but it makes the code more readable.

Reply:
Is there a preferred way to do checks for multiple flags?

The reason I used the above was that I could do:

	if (!(svc->flags & (SVC_CHECK_LOADBALANCER |
			    SVC_CHECK_NODEPORT |
			    SVC_CHECK_EXTERNAL_IP |
			    SVC_CHECK_HOSTPORT))) {
		return DROP_IS_CLUSTER_IP;
	}

With using helpers it would be similar as the old code, which checks each bit individually and leaves it up to the compiler to try and improve it:

	if (!svc || (!lb4_svc_is_loadbalancer(svc) &&
		     !lb4_svc_is_nodeport(svc) &&
		     !lb4_svc_is_external_ip(svc) &&
		     !lb4_svc_is_hostport(svc))) {
		if (svc)
			return DROP_IS_CLUSTER_IP;